aws-athena/views/model-vw_pin_shared_input.sql

```diff
@@ -103,6 +103,32 @@ school_district_ratings AS (
         COUNT(*) AS num_schools_in_district
     FROM {{ source('other', 'great_schools_rating') }}
     GROUP BY district_geoid, district_type
+),
+
+exemption_features AS (
+    SELECT
+        year,
+        pin,
+        SIGN(exe_homeowner) AS ccao_is_active_exe_homeowner,
+        act AS ccao_n_years_exe_homeowner
+    FROM (
+        SELECT
+            *,
+            SUM(SIGN(exe_homeowner))
+                OVER (PARTITION BY pin, grp ORDER BY year)
```

Comment:
**[Question, non-blocking]** I don't totally get what the `grp` subquery is doing for us here. Without a detailed understanding of the underlying data model, my naive assumption would be that `SUM(SIGN(exe_homeowner)) OVER (PARTITION BY pin ORDER BY year)` would be enough to get the history of the `exe_homeownrer` column. Is this the "down-up fill" you mention in the comment above?

Reply:
It's not intuitive at all, but the purpose of `grp` is to reset the cumulative count of years whenever the `exe_homeowner` value is 0 (i.e. whenever someone loses the exemption).